### PR TITLE
refactor: remove ctx from next

### PR DESCRIPTION
**Pull request by craigpastro**
## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/storage/iterators.go [hunk 1]
```diff
@@ -58,11 +57,11 @@ func NewUniqueObjectIterator(iter1, iter2 ObjectIterator) ObjectIterator {
 
 var _ ObjectIterator = (*uniqueObjectIterator)(nil)
 
-// Next returns the next most unique object from the two underlying iterators.
-// If the context is cancelled or times out, it should return ErrIteratorDone
-func (u *uniqueObjectIterator) Next(ctx context.Context) (*openfgapb.Object, error) {
+// Next returns the next unique object from the two underlying iterators. If
+// the context is cancelled or times out, it should return ErrIteratorDone
```
Comment (jon-whit) [inline pkg/storage/iterators.go:61]: The bit about Context here is no longer needed in the godoc.

**Review by jon-whit — CHANGES_REQUESTED** [review]
Couple of minor changes requested. Other than that, looks good 👍

File: pkg/storage/iterators.go (hunk 1 above)
Comment (craigpastro) [reply pkg/storage/iterators.go]: Good catch. Fixed. Thanks!

File: pkg/storage/iterators.go [hunk 2]
```diff
@@ -13,7 +12,7 @@ var ErrIteratorDone = errors.New("iterator done")
 
 type Iterator[T any] interface {
 	// Next will return the next available item. If the context is cancelled or times out, it should return ErrIteratorDone
```
[the remark this answers is not in the document]
Comment (craigpastro) [reply pkg/storage/iterators.go]: Good catch. Fixed. Thanks!

**Review by jon-whit — APPROVED** [review]

**Pull request merged**